src/convertToObject.js

```diff
@@ -3,16 +3,32 @@
 /**
  * Implement convertToObject function:
  *
- * Function takes a string with styles (see an example in [stylesString.js](./stylesString.js))
+ * Function takes a string with styles
+ * (see an example in [stylesString.js](./stylesString.js))
  * and returns an object where CSS properties are keys
- * and values are the values of related CSS properties (see an exampl in [test file](./convertToObject.test.js))
+ * and values are the values of related CSS properties
+ * (see an exampl in [test file](./convertToObject.test.js))
  *
  * @param {string} sourceString
  *
  * @return {object}
  */
 function convertToObject(sourceString) {
-  // write your code here
+  const originString = sourceString.replace(/\s\s+/g, ' ');
+  const styles = originString.split(';');
+  const stylesToObject = {};
+  const validStyles = styles
+    .filter(style => style
+      .split('')
+      .find(symbol => symbol === ':'));
+
+  validStyles.map((style) => {
+    const copyStyle = style.split(':');
+
+    stylesToObject[copyStyle[0].trim()] = copyStyle[1].trim();
+  });
```

Comment:
don't use map to mutate anything, map is needed to transform array elements and create new one. use reduce to build an object